org.civicrm.reqangular/src/common/controllers/dialog/dialog-ctrl.js

```diff
@@ -1,29 +1,61 @@
-define([], function () {
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'common/modules/controllers'
+], function (_, controllers) {
   'use strict';
 
-  return ['$q', '$scope','$uibModalInstance', 'content', 'onConfirm',
-    function ($q, $scope, $modalInstance, content, onConfirm) {
-      $scope.loading = false
-
-      $scope.title = content.title || 'CiviHR';
-      $scope.msg = content.msg || '';
-      $scope.copyConfirm = content.copyConfirm || 'Yes';
-      $scope.copyCancel = content.copyCancel || 'Cancel';
-      $scope.classConfirm = content.classConfirm || 'btn-primary';
-
-      $scope.confirm = function () {
-        $scope.loading = true;
-
-        $q
-          .resolve(onConfirm ? onConfirm() : null)
-          .then(function () {
-            $modalInstance.close(true);
-          });
-      };
-
-      $scope.cancel = function () {
-        $modalInstance.close(false);
-      };
+  controllers.controller('DialogController', DialogController);
+
+  DialogController.$inject = ['$q', '$scope', '$uibModalInstance', 'props'];
+
+  function DialogController ($q, $scope, $modalInstance, props) {
+    $scope.cancel = cancel;
+    $scope.confirm = confirm;
+
+    (function init () {
+      assignProps(props);
+      props.delayedProps && props.delayedProps().then(function (_props_) {
+        assignProps(_props_);
+      });
+    }());
+
+    /**
+     * Assignes passed properties to the $scope,
+     * uses default values for some of properties
+     *
+     * @param {Object} props
+     */
+    function assignProps (props) {
+      _.assign($scope, _.defaultsDeep(props, {
+        title: 'CiviHR',
+        msg: '',
+        copyConfirm: '',
+        copyCancel: '',
```

Comment:
Why don't we set cancel and confirm default values?

Reply:
We don't want to change the default values since this can affect some places that expect exactly these default values.